### PR TITLE
ELEMENTS-1393: filter out non visible inputs for layout validation

### PR DESCRIPTION
**Pull request by richardsd**

### CODE REVIEW

**Review by mnixo — APPROVED** [review]
In the following scenario:
- A create layout with several widget fields.
- A required dc:title input widget on top.
- Scrolling down to the bottom of the layout, "hiding" the dc:title widget. 

The required dc:title widget is still validated, right?

**Comment by richardsd** [comment]
> In the following scenario:
> 
> * A create layout with several widget fields.
> * A required dc:title input widget on top.
> * Scrolling down to the bottom of the layout, "hiding" the dc:title widget.
> 
> The required dc:title widget is still validated, right?

Yes, it does. I'll upload a video with that behavior too.

File: ui/nuxeo-layout.js [hunk 1]
```diff
@@ -149,6 +149,10 @@ import './nuxeo-error.js';
       return submittable;
     }
 
+    _isVisible(node) {
+      return node && node.offsetParent && node.offsetHeight > 0 && node.offsetWidth > 0;
```
Comment (Gabez0r) [inline ui/nuxeo-layout.js:153]: Looks good to me. I think we should leave a comment of why we are using this tricky: we want to exclude fallback content from slots (and also point to ELEMENTS-1393 for more details).

**Comment by Gabez0r** [comment]
Could you please run the legacy pipeline on this? 🔵

**Comment by richardsd** [comment]
> Could you please run the legacy pipeline on this? 🔵

If/When the pipeline is fixed, I'll give it a try 👍

**Comment by richardsd** [comment]
@mnixo is this the behavior you were describing?
elements-1393

**Comment by mnixo** [comment]
> @mnixo is this the behavior you were describing?

Exactly 👍

**Review by mnixo — APPROVED** [review]

**Comment by richardsd** [comment]
> Could you please run the legacy pipeline on this? 🔵

Legacy pipeline here 🔵 https://qa.nuxeo.org/jenkins/job/web-ui-pipeline-rdias/152/

**Review by Gabez0r — APPROVED** [review]

**Pull request merged**